uclibc does not support _DATE_FMT, it seems to be a libc only thing
not sure this will work, will need to take a closer look.
http://pubs.opengroup.org/onlinepubs/007904975/basedefs/langinfo.h.html
http://sources.redhat.com/ml/libc-alpha/2000-10/msg00298.html

--- ./lib/libspl/timestamp.c.orig
+++ ./lib/libspl/timestamp.c
@@ -40,7 +40,7 @@
 
 	/* We only need to retrieve this once per invocation */
 	if (fmt == NULL)
-		fmt = nl_langinfo(_DATE_FMT);
+		fmt = nl_langinfo(D_FMT);
 
 	if (timestamp_fmt == UDATE) {
 		(void) printf("%ld\n", t);
